x-pack/plugins/apm/public/components/app/metrics/index.tsx

```diff
@@ -6,17 +6,30 @@
  */
 
 import React from 'react';
-import { isJavaAgentName, isJRubyAgent } from '../../../../common/agent_name';
+import {
+  isJavaAgentName,
+  isJRubyAgent,
+  isServerlessAgent,
+} from '../../../../common/agent_name';
 import { useApmServiceContext } from '../../../context/apm_service/use_apm_service_context';
+import { ServerlessMetrics } from './serverless_metrics';
 import { ServiceMetrics } from './service_metrics';
 import { JvmMetricsOverview } from './jvm_metrics_overview';
 
 export function Metrics() {
   const { agentName, runtimeName } = useApmServiceContext();
+  const isServerless = isServerlessAgent(runtimeName);
 
-  if (isJavaAgentName(agentName) || isJRubyAgent(agentName, runtimeName)) {
+  if (
+    !isServerless &&
+    (isJavaAgentName(agentName) || isJRubyAgent(agentName, runtimeName))
+  ) {
     return <JvmMetricsOverview />;
   }
 
+  if (isServerless) {
+    return <ServerlessMetrics />;
+  }
```

Comment:
Move this to the top. Then you don't have to check for `!isServerless` for `JvmMetricsOverview`

Reply:
This was actually intended. I decided to have both checks to not rely on the positions of the if-clauses. I'd rather have extra check than create a new bug if we accidentally move it.

Reply:
fair enough 👍